app/src/main/java/com/pajato/android/gamechat/exp/chess/ChessEngine.java

```diff
@@ -337,24 +366,25 @@ private void promotePawn(final int position, final Team team) {
     }
 
     /** Remove any possible moves that would result in placing the active King in check. */
-    private void removeCheckExposingMoves(final List<Integer> possibleMoves) {
+    private void removeCheckExposingMoves(final List<Integer> possibleMoves, final int piecePosition) {
         // Collect the rejected moves for removal after all possible positions have been tested.
         List<Integer> exposedMoves = new ArrayList<>();
         for (int possiblePosition : possibleMoves)
-            if (testForCheck(possiblePosition))
+            if (testForCheck(possiblePosition, piecePosition))
                 exposedMoves.add(possiblePosition);
         for (int position : exposedMoves)
             possibleMoves.remove(Integer.valueOf(position));
     }
 
     /** Return true if moving the selected piece to the given position would expose a check. */
-    private boolean testForCheck(final int position) {
+    private boolean testForCheck(final int position, final int selectedPosition) {
         // Move the selected piece to the possible move and test to see if the associated King
-        // would be in check.  First, save any displaced piece information.
+        // would be in check.  First, save any displaced piece information. Moving nothing cannot
+        // put the player in check, so if there is no piece at the position, then return false.
         boolean result;
         ChessBoard board = mModel.board;
-        int savedSelectedPosition = board.getSelectedPosition();
-        ChessPiece savedSelectedPiece = board.delete(savedSelectedPosition);
+        ChessPiece savedSelectedPiece = board.delete(selectedPosition);
+        if(savedSelectedPiece == null) return false;
```

Comment:
We have decided that it is better to put the conditional statement on a separate line to facilitate setting breakpoints.  We still don't like the procedural if style: if(condition) but if (condition) is fine.